feat(components): [form-item]: add `label-position` prop (#17111)
element-plus/element-plus: docs/en-US/component/form.md

## Alignment

> Depending on your design, there are several different ways to align your label element. You can set the position of `el-form-item` separately using `label-position`. If the value is not set, the `label-position` of `el-form` is used.

Reviewer: ```suggestion
Depending on your design, there are several different ways to align your label element. You can set the position of `el-form-item` separately using `label-position` ^(2.7.7). If the value is not set, the `label-position` of `el-form` is used.
```

:::

> ## Custom Form Item labelPosition

Reviewer 2: Please add a version tag.
```suggestion
## Custom Form Item labelPosition ^(2.7.6)
```

> The label of the component can be positioned using the `label-position` property in el-form-item. The default extend `el-form`

Reviewer: ```suggestion
You can set the position of `el-form-item` separately using `label-position`. If the value is not set, the `label-position` of `el-form` is used.
```

## FormItem API

### FormItem Attributes

| Name                    | Description                                                                                                                                                            | Type                                                | Default |
| ----------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------- | ------- |
| prop                    | A key of `model`. It could be a path of the property (e.g `a.b.0` or `['a', 'b', '0']`). In the use of `validate` and `resetFields` method, the attribute is required. | ^[string] / ^[string&#91;&#93;]                     | —       |
| label                   | Label text.                                                                                                                                                            | ^[string]                                           | —       |

> | label-position ^(2.7.6) | Position of item label. If set to `'left'` or `'right'`, `label-width` prop is also required. Default extend `label-postion` of `form`.                                | ^[enum]`'left' \| 'right' \| 'top'`                 | ''      |

Reviewer: ```suggestion
| label-position ^(2.7.7) | Position of item label. If set to `'left'` or `'right'`, `label-width` prop is also required. Default extend `label-postion` of `form`.                                | ^[enum]`'left' \| 'right' \| 'top'`                 | ''      |
```

| --------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------- | ------- |
| prop            | A key of `model`. It could be a path of the property (e.g `a.b.0` or `['a', 'b', '0']`). In the use of `validate` and `resetFields` method, the attribute is required. | ^[string] / ^[string&#91;&#93;]                     | —       |
| label           | Label text.                                                                                                                                                            | ^[string]                                           | —       |

> | label-position  | Position of item label. If set to `'left'` or `'right'`, `label-width` prop is also required. Default extend `label-postion` of `form`                                 | ^[enum]`'left' \| 'right' \| 'top'`                 | ''      |

Reviewer 2: ```suggestion
| label-position ^(2.7.6)  | Position of item label. If set to `'left'` or `'right'`, `label-width` prop is also required. Default extend `label-postion` of `form`                                 | ^[enum]`'left' \| 'right' \| 'top'`                 | ''      |
```

> | label-position ^(2.7.6)  | Position of item label. If set to `'left'` or `'right'`, `label-width` prop is also required. Default extend `label-postion` of `form`                                 | ^[enum]`'left' \| 'right' \| 'top'`                 | ''      |

Reviewer 2: ```suggestion
| label-position ^(2.7.6)  | Position of item label. If set to `'left'` or `'right'`, `label-width` prop is also required. Default extend `label-postion` of `form`.                                 | ^[enum]`'left' \| 'right' \| 'top'`                 | ''      |
```
The document should be formatted to align with the other fields.
